File: Ejercicios/GO/Clases/Libro.go
package main

type Libro struct {
	Titulo string
	Autor  string
}

func NewLibro() Libro {
	return Libro{Titulo: "Sin título", Autor: "Anónimo"}
}

func NewLibroConDatos(titulo, autor string) Libro {
	return Libro{Titulo: titulo, Autor: autor}
}

func NewLibroTitulo(titulo string) Libro {
	return Libro{Titulo: titulo, Autor: "Desconocido"}
}

func NewLibroCopy(l Libro) Libro {
	return Libro{Titulo: l.Titulo, Autor: l.Autor}
}

File: Ejercicios/JavaScript/Clases/Libro.js
class Libro {
  constructor(titulo, autor) {
    if (titulo === undefined && autor === undefined) {
      this.titulo = "Sin título";
      this.autor = "Anónimo";
    } else if (titulo instanceof Libro) {
      this.titulo = titulo.titulo;
      this.autor = titulo.autor;
    } else if (autor === undefined) {
      this.titulo = titulo;
      this.autor = "Desconocido";
    } else {
      this.titulo = titulo;
      this.autor = autor;
    }
  }
}

export default Libro;
